Sources/Starknet/Crypto/StarknetPoseidon.swift

```diff
@@ -45,12 +45,6 @@ public class StarknetPoseidon {
     ///     - elements: array of Felt values to hash.
     /// - Returns: Poseidon hash of the values as Felt.
     public class func poseidonHash(_ values: [Felt]) -> Felt {
-        if values.count == 1 {
-            return poseidonHash(values[0])
-        }
-        if values.count == 2 {
-            return poseidonHash(first: values[0], second: values[1])
-        }
```

Comment:
what's the reason behind this?

Reply:
They produce different hash than the list approach.

I thought these are edge cases, but they aren't. We actually need to use list approach even on single/two values when computing v3 hashes.